training/TRAINING.md

```diff
@@ -0,0 +1,213 @@
+
+### 2. Initialize a model from scratch or use a pre-trained one.
+
+Depending on your compute resources and your dataset, you need to choose between fine-tuning a pre-trained model and training a new model from scratch.
+
+In that sense, we released a 300M checkpoint trained on 10.5K hours of annotated data under the repository id: [`parler-tts/parler_tts_300M_v0.1`](https://huggingface.co/parler-tts/parler_tts_300M_v0.1), that you can fine-tune for your own use-case.
+
+You can also train you own model from scratch. You can find [here](/helpers/model_init_scripts/) examples on how to initialize a model from scratch. For example, you can initialize a dummy model with:
+
+```sh
+python helpers/model_init_scripts/init_dummy_model.py ./parler-tts-untrained-dummy --text_model "google-t5/t5-small" --audio_model "parler-tts/dac_44khZ_8kbps"
+```
+
+In the rest of this guide, we'll use a 300-M parameters that we'll initialize with:
+
+```sh
+python helpers/model_init_scripts/init_model_300M.py ./parler-tts-untrained-300M --text_model "google/flan-t5-base" --audio_model "parler-tts/dac_44khZ_8kbps"
+```
+
+
+### 3. Create or find datasets
+
+To train your own Parler-TTS, you need datasets with 3 main features:
+- speech data
+- text transcription of the speech data
+- conditionning text description - that you can create using [Data-Speech](https://github.com/huggingface/dataspeech), a library that allows you to annotate the speaker and utterance characteristics with natural language description.
+
+Note that we made the choice to use description of the main speech characteristics (speaker pitch, speaking rate, level of noise, etc.) but that you are free to use any handmade or generated text description that makes sense.
+
+In the rest of this guide, and to make it simple, we'll use the [4.8K-samples clean test split](https://huggingface.co/datasets/blabble-io/libritts_r/viewer/clean/test.clean) of [LibriTTS-R](https://huggingface.co/datasets/blabble-io/libritts_r/). We've annotated LibriTTS-R using [Data-Speech](https://github.com/huggingface/dataspeech) and shared the resulting dataset here: [parler-tts/libritts_r_tags_tagged_10k_generated](https://huggingface.co/datasets/parler-tts/libritts_r_tags_tagged_10k_generated).
+
+
+## C. Training
+
+The script [`run_parler_tts_training.py`](/training/run_parler_tts_training.py) is an end-to-end script that:
+1. load dataset(s) and merge them to the annotation dataset(s) if necessary
+2. pre-compute audio tokens
+3. train Parler-TTS
+
+In this example, we will train and evaluate on a subsample of the test split. This is purely to demonstrate how to use the training script, rather than recommended advice for defining train/validation splits. We advise that you train on the train splits of your dataset, evaluate and tune hyper-parameters on the validation split, and only test the final checkpoint on the test split.
```

Comment:
From experience it's better to provide an example that actually gets reasonable results. I've tried in the past giving lower resource dummy examples, and what ends up happening is that some users can't figure out how to adapt them for actual training runs -> my advice here would be to include the args that will give proper training results, ideally those that were used to train the v0.1 model. This way, people can build on your work most easily